### PR TITLE
feat(core): add open link in app to doc menu

### PR DESCRIPTION
**Pull request by pengx17**
add "open in desktop app" menu item for editor

fix AF-1547

### CODE REVIEW

**Comment by pengx17** [comment]
* **#8630** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8630?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8601** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8601?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8597** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8597?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#8588** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8588?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pengx17 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/modules/open-in-app/utils.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  }
+
+  const urlObject = new URL(url);
+  const params = urlObject.searchParams;
+
+  if (newTab) {
+    params.set('new-tab', '1');
+  }
+
+  return `${scheme}://${urlObject.hostname}${urlObject.pathname}?${params.toString()}#${urlObject.hash}`;
```
Comment (CatsJuice) [inline packages/frontend/core/src/modules/open-in-app/utils.ts]: Is it unnecessary to handle `port` | `empty search params` | `empty hash`?

Comment (pengx17) [reply packages/frontend/core/src/modules/open-in-app/utils.ts]: port is part of hostname;
if search params is empty we will have:
`affine://localhost:8080?#ssss`, which is a valid case

Comment (CatsJuice) [reply packages/frontend/core/src/modules/open-in-app/utils.ts]: 🤔 `port` is included in `host` but not `hostname`?

Comment (pengx17) [reply packages/frontend/core/src/modules/open-in-app/utils.ts]: indeed. my bad. thanks for pointing out!

**Comment by pengx17** [comment]
### Merge activity

* **Oct 28, 2:17 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/8597) disabled "merge when ready" on this PR due to: a merge conflict with the target branch; resolve the conflict and try again..
* **Oct 29, 1:49 AM EDT**: A user added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=toeverything&repo=AFFiNE).
* **Oct 29, 2:11 AM EDT**: A user  merged this pull request with the [Graphite merge queue](https://app.graphite.dev/merges?org=toeverything&repo=AFFiNE).

File: packages/frontend/apps/electron/src/main/deep-link.ts [hunk 2]
```diff
@@ -81,6 +82,23 @@ async function handleAffineUrl(url: string) {
       method,
       payload,
     });
+  } else if (
+    urlObj.searchParams.get('new-tab') &&
+    urlObj.pathname.startsWith('/workspace')
+  ) {
+    // basename of /workspace/xxx/yyy is /workspace/xxx
+    const basename = urlObj.pathname.split('/').slice(0, 3).join('/');
```
Comment (forehalo) [inline packages/frontend/apps/electron/src/main/deep-link.ts:91]: add a TODO for me, we need to refactor routes utilities

**Review by forehalo — APPROVED** [review]

**Pull request merged**